### PR TITLE
[GeoMechanicsApplication] Fixed invalid values for default linear solver settings

### PR DESCRIPTION
**Pull request by rfaasse**

### CODE REVIEW

File: applications/GeoMechanicsApplication/python_scripts/geomechanics_Pw_solver.py [hunk 1]
```diff
@@ -75,15 +75,14 @@ def GetDefaultParameters(cls):
...
-                "krylov_type": "gmres",
-                "coarsening_type": "aggregation"
+                "solver_type":     "amgcl",
+                "smoother_type":   "ilu0",
+                "krylov_type":     "gmres",
+                "coarsening_type": "aggregation",
+                "max_iteration":   100,
+                "verbosity":       0,
+                "tolerance":       1.0e-6,
+                "scaling":         false
```
Comment (rfaasse) [inline applications/GeoMechanicsApplication/python_scripts/geomechanics_Pw_solver.py]: Of course we can discuss if these are the best defaults we can think of, I just took the settings of one of our tests. I first tried sparse_lu, but I don't think it's always guaranteed that the LinearSolversApplication is initialized, so not sure if that's the most robust option for a default

File: applications/GeoMechanicsApplication/python_scripts/geomechanics_Pw_solver.py [hunk 2]
```diff
@@ -75,15 +75,14 @@ def GetDefaultParameters(cls):
...
-                "scaling": false,
-                "verbosity": 0,
-                "preconditioner_type": "ILU0Preconditioner",
-                "smoother_type": "ilu0",
-                "krylov_type": "gmres",
-                "coarsening_type": "aggregation"
+                "solver_type":     "amgcl",
+                "smoother_type":   "ilu0",
+                "krylov_type":     "gmres",
+                "coarsening_type": "aggregation",
```
Comment (WPK4FEM) [inline applications/GeoMechanicsApplication/python_scripts/geomechanics_Pw_solver.py]: The precconditioner_type is missing here. Maybe it is wise to do the lower case of AMGCL as only change.

**Review by WPK4FEM — COMMENTED** [review]
Hi Richard,
I'm probably missing some information here. Your remark was about uppercase not being accepted for AMGCL in the solver type. I don't understand why then all other linear solver settings should be rearranged and the preconditioner_type removed. Preconditioner type means something to me, not all of the other settings ( but that is a gap in my knowledge ).
Regards, Wijtze Pieter

File: applications/GeoMechanicsApplication/python_scripts/geomechanics_Pw_solver.py (hunk 2 above)
Comment (rfaasse) [reply applications/GeoMechanicsApplication/python_scripts/geomechanics_Pw_solver.py]: Agreed, I reverted everything that wasn't related, just did the lower case amgcl, however then we get the following problem:
```
RuntimeError: Error: AMGCL Linear Solver : preconditioner_type is invalid!
Currently prescribed preconditioner_type : ILU0Preconditioner
Admissible values are :
    amg
    dummy
    relaxation
```

So I changed that to amg

**Review by WPK4FEM — APPROVED** [review]
The minimal change and now tested.
Nice.

**Pull request merged**